docs/content/en/docs/crd-ref/TaskDefinition/index.md

```diff
@@ -0,0 +1,279 @@
+---
+title: KeptnTaskDefinition
+description: Define tasks that can be run pre- or post-deployment
+weight: 89
+---
+
+A `KeptnTaskDefinition` defines tasks
+that are run by the Keptn Lifecycle Toolkit
+as part of the pre- and post-deployment phases of a `KeptnApp`.
+
+## Yaml Synopsis
+
+```yaml
+apiVersion: <lifecycle.keptn.sh/v?alpha?>
+kind: KeptnTaskDefinition
+metadata:
+  name: <task-name>
+spec:
+  function:
+    inline: | httpRef | functionRef
+    [parameters:
+      map:
+        textMessage: "This is my configuration"]
+    [secureParameters:
+      secret: slack-token]
+```
+
+## Fields
+
+* **apiVersion** -- API version being used.
+`
+* **kind** -- CRD type.  Must be set to `KeptnTaskDefinition`
+
+* **name** -- Unique name of this task.
+  * Must be an alphanumeric string and, by convention, is all lowercase.
+  * Can include the special characters `_`, `-`, <what others>.
+  * Should not include spaces.
+
+* **function** -- Code to be executed,
+  expressed as a [Deno](https://deno.land/) script.
+  Refer to [function runtime](https://github.com/keptn/lifecycle-toolkit/tree/main/functions-runtime)
+  for more information about the runtime.
+  In the future, we intend to support additional runtimes,
+  especially running a container image directly.
+
+  The `function` can be defined as one of the following:
+
+  * **inline** - Include the actual executable code to execute.
+    This can be written as a full-fledged Deno script
+    that is included in this file.
+    For example:
+
+    ```function:
+    inline:
+      code: |
+        console.log("Deployment Task has been executed");
+    ```
+
+  * **httpRef** - Specify a Deno script to be executed at runtime
+    from the remote webserver that is specified.
+    For example:
+
+    ```yaml
+    name: hello-keptn-http
+      spec:
+        function:
+          httpRef:
+            url: <url>
+    ```
+
+  * **functionRef** -- Execute another `KeptnTaskDefinition` that has been defined.
+    Populate this field with the value of the `name` field
+    for the `KeptnTaskDefinition` to be called.
+    This is commonly used to call a general function
+    that is used in multiple places, possibly with different parameters.
+    An example is:
+
+     ```yaml
+     spec:
+       function:
+         functionRef:
+           name: slack-notification
+     ```
+
+    This can also be used to group a set of tasks
+    into a single `KeptnTaskDefinitions`,
+    such as defining a `KeptnTaskDefinition` for testing.
+    In this case, it other, existing `KeptnTaskDefinition`s
+    for each type of test to be run,
+    specifying each by the value of the `name` field.
+
+* **parameters** - An optional field to supply input parameters to a function.
+  The Lifecycle Toolkit passes the values defined inside the `map` field
+  as a JSON object.
+  For example:
+
+   ```spec:
+       parameters:
+         map:
+           textMessage: "This is my configuration"
+   ```
```

Comment:
```suggestion
   ```yaml
   spec:
     parameters:
       map:
         textMessage: "This is my configuration"
   ```
```